engineering/adr-0002-height-and-round-in-unlock-tx-payload.md

```diff
@@ -0,0 +1,101 @@
+in platform state and logic in Drive implemented to support this solution.
+
+There two ways to proceed with credits withdrawal implementation:
+1. Add height and round to asset lock transaction payload and remove unnecessary in this case "two block commit" solution to simplify the system
+2. Change tenderdash to sign the whole transaction instead of DIP7 format.
+
+Both approaches have benefits and downsides. In this ADR we will look closer to the fist solution.
+There is [ADR0003](./adr-0003-vote-extensions-without-DIP7.md) that is dedicated to the second approach.
+
+## Decision
+
+To allow asset unlock transaction consumers verify signature, concatenated prefix `ve` (Dash Platform Vote Extension), platform height (uint64)
+and platform round (uint32) as `signatureRequest` (14 bytes) field should be added to asset unlock transaction payload. Having a complete request
+message as part of the transaction payload will allow not just tenderdash but other L2 systems to utilize Asset Lock/Unlock Core functionality for
+their needs in the future.
+
+To verify the signature, asset unlock transaction consumer should hash `signatureRequest` from payload a combine with `quorumHash` and `sigHash`
+according to DIP7:
+
+```
+requestID = SHA256(signatureRequest from transaction payload)
+
+sigHash = SHA256(transaction without quorumSig)
+
+SignID = SHA256(quorumHash, requestID, sigHash)
+```
+
+The two block commit logic that mitigates "multi round signature shares" attack using additional step with a queue become unnecessary and can be removed from the codebase to simplify the withdrawal process.
+
+An additioanl security layer could be added to [Sign](https://dashcore.readme.io/docs/core-api-ref-remote-procedure-calls-evo#quorum-sign) Core's RPC.
+List of quorums types should be limited to platform only. It will guarantiee that Sign RPC method won't be used to attack Layer 1 functionality like
+Instant Send and Chain Lock. 
+
+
+## Implementation
+
+* Revert commit with additional asset lock payload types form rust-dashcore library (Platform team)
+* Remove withdrawal transactions queue and related logic from Drive (Platform team)
+* Add `requestID` to Asset Unlock Payload and update signature verification function in rust-dashcore (Platform team)
+* Add `requestID` to Asset Unlock Payload and update signature verification function in Core (Core team)
+* Change request ID prefix to `ve` in Tenderdash (Tenderdash team)
+
+## Consequences
+
+### Pros
+
+* An Asset Unlock transaction will be signed and broadcasted to Core on the same block when the withdrawal was requested.
+  It speeds up withdrawal up to 3 minutes (empty block interval).
+* Simpler withdrawal functionality. Unnecessary logic, data and IO operations will be removed that will simplify
+  the system and reduce the load.
+
+### Cons
+
+* Additional 16 bytes will be added to Asset Unlock transaction
```

Comment:
-* Additional 16 bytes will be added to Asset Unlock transaction
+* Additional 14 bytes will be added to Asset Unlock transaction